File: .API/Cloud/AssetInfo.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.AssetInfo
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 2635624C-5F24-4EFB-ACD1-7E9C1349B2F5
// Assembly location: F:\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
  public class AssetInfo
  {
    [JsonProperty(PropertyName = "ownerId")]
    [JsonPropertyName("ownerId")]
    public string OwnerId { get; set; }

    [JsonProperty(PropertyName = "assetHash")]
    [JsonPropertyName("assetHash")]
    public string AssetHash { get; set; }

    [JsonProperty(PropertyName = "bytes")]
    [JsonPropertyName("bytes")]
    public long Bytes { get; set; }

    [JsonProperty(PropertyName = "free")]
    [JsonPropertyName("free")]
    public bool Free { get; set; }

    [JsonProperty(PropertyName = "isUploaded")]
    [JsonPropertyName("isUploaded")]
    public bool IsUploaded { get; set; }

    [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "uploaderUserId")]
    [JsonPropertyName("uploaderUserId")]
    public string UploaderUserId { get; set; }

    [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "countsAgainstMemberQuota")]
    [JsonPropertyName("countsAgainstMemberQuota")]
    public bool? CountsAgainstMemberQuota { get; set; }
  }
}

File: src/api/AssetInfo.ts
/** AssetInfo.
 * @class AssetInfo
 */
export class AssetInfo {
	public OwnerId: string;
	public AssetHash: string;
	public Bytes: number;
	public Free: number;
	public IsUploaded: boolean;
	public UploaderUserId?: string;
	public CountsAgainstMemberQuota?: boolean;
	constructor($b: AssetInfoJSON) {
		this.OwnerId = $b.ownerId;
		this.AssetHash = $b.assetHash;
		this.Bytes = $b.bytes;
		this.Free = $b.free;
		this.IsUploaded = $b.isUploaded;
		this.UploaderUserId = $b.uploaderUserId;
		this.CountsAgainstMemberQuota = $b.countsAgainstMemberQuota;
	}
	toJSON(): AssetInfoJSON {
		return {
			ownerId: this.OwnerId,
			assetHash: this.AssetHash,
			bytes: this.Bytes,
			free: this.Free,
			isUploaded: this.IsUploaded,
			uploaderUserId: this.UploaderUserId,
			countsAgainstMemberQuota: this.CountsAgainstMemberQuota,
		};
	}
}
interface AssetInfoJSON {
	ownerId: string;
	assetHash: string;
	bytes: number;
	free: number;
	isUploaded: boolean;
	uploaderUserId?: string;
	countsAgainstMemberQuota?: boolean;
}
